src/kademlia/dht_tracker.cpp

```diff
@@ -117,14 +117,16 @@ namespace libtorrent { namespace dht {
 	{
 		address local_address = s->get_local_endpoint().address();
 #if TORRENT_USE_IPV6
-		// don't try to start dht nodes on non-global addresses
+		// don't try to start dht nodes on non-global IPv6 addresses
+		// with IPv4 the interface might be behind NAT so we can't skip them based on the scope of the local address
+		// and we might not have the external address yet
```

Comment:
That logic will be going into the last patch in the series, but it still won't listen on loopback by default. Now that I think about it that may confuse people trying to test transfers locally because `localhost` often resolves to ::1 these days. I may have to modify the final patch to disallow DHT and tracker announces on loopback.